### PR TITLE
upgrade kyber swap role contract

### PR DESCRIPTION
**Pull request by Zohar97**

### CODE REVIEW

File: contracts/base/KyberSwapRole.sol [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    require(operators[msg.sender], 'KyberSwapRole: not operator');
+    _;
+  }
+
+  modifier onlyGuardian() {
+    require(guardians[msg.sender], 'KyberSwapRole: not guardian');
+    _;
+  }
+
+  modifier onlyEnabled() {
```
Comment (thongtrungtran) [inline contracts/base/KyberSwapRole.sol]: I think we should just name it like `paused` or `unpaused`, better understand

Comment (manhlx3006) [reply contracts/base/KyberSwapRole.sol]: yeah paused or unpaused is also ok

File: contracts/base/KyberSwapRole.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    require(isLogicEnabled, 'KyberSwapRole: not enabled');
+    _;
+  }
+
+  constructor() {
+    owner = msg.sender;
```
Comment (manhlx3006) [inline contracts/base/KyberSwapRole.sol]: Why not using Ownable for this?

File: contracts/base/KyberSwapRole.sol [hunk 3]
```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
```
Comment (manhlx3006) [inline contracts/base/KyberSwapRole.sol]: Please also add some comments to codes as well

File: contracts/KSElasticLMV2.sol [hunk 4]
```diff
@@ -7,15 +7,15 @@ import {ReentrancyGuard} from '@openzeppelin/contracts/security/ReentrancyGuard.
 import {EnumerableSet} from '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
```
Comment (manhlx3006) [inline contracts/KSElasticLMV2.sol:7]: Let's not use paused and unpaused here, should just a PR to update KyberSwapRole first.
I think only allowing to claim rewards/fees or withdraw liquidity when unpaused is not a good design (shouldn't block users from these actions)

**Review by vutt94 — APPROVED** [review]

**Pull request merged**